src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/client/TransactionsServiceClient.kt

```diff
@@ -0,0 +1,40 @@
+package it.pagopa.ecommerce.eventdispatcher.client
+
+import it.pagopa.ecommerce.eventdispatcher.exceptions.BadGatewayException
+import it.pagopa.ecommerce.eventdispatcher.exceptions.GatewayTimeoutException
+import it.pagopa.ecommerce.eventdispatcher.exceptions.TransactionNotFound
+import it.pagopa.generated.transactionauthrequests.v1.api.TransactionsApi
+import it.pagopa.generated.transactionauthrequests.v1.dto.TransactionInfoDto
+import it.pagopa.generated.transactionauthrequests.v1.dto.UpdateAuthorizationRequestDto
+import java.util.*
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import reactor.core.publisher.Mono
+
+@Component
+class TransactionsServiceClient(
+  @Autowired @Qualifier("transactionsServiceWebClient") private val transactionsApi: TransactionsApi
+) {
+
+  fun patchAuthRequest(
+    transactionId: UUID,
+    updateAuthorizationRequestDto: UpdateAuthorizationRequestDto
+  ): Mono<TransactionInfoDto> {
+    return transactionsApi
+      .updateTransactionAuthorization(transactionId.toString(), updateAuthorizationRequestDto)
+      .onErrorMap(WebClientResponseException::class.java) { exception: WebClientResponseException ->
+        when (exception.statusCode) {
+          HttpStatus.BAD_REQUEST -> TransactionNotFound(transactionId)
```

Comment:
maybe exception mapping have to be reviewed, f.e. 401 -> TransactionNotFound, is right?